nemo_aligner/algorithms/critic_server_trainer.py

```diff
@@ -217,23 +251,11 @@ def run_inference(self, inputs=None):
         """only rank 0 has valid data
```

Comment:
the input data is still available only on rank 0, but every rank has to call run distributed inference because of all the broadcasting logic.

I've fixed the comment here: ad40b44b251d43abac466230fe0d53f82f1d4b25

Reply:
Ah I think there was some confusion -- I was talking about output data and I thought that's what the comment was referring to. In the past I believe that only rank 0 had valid output data because we were gathering on rank 0 only. But in your new implementation, the output is available to all ranks (based on my understanding).

Can you just update the comment to:
```python
"""only rank 0 needs valid input data, but all other ranks should call `run_inference()`
"""
```

Reply:
ah I see, yeah you're right. So in the implementation on main we use gather_tensor https://github.com/NVIDIA/NeMo-Aligner/blob/main/nemo_aligner/models/nlp/gpt/megatron_gpt_reward_model.py#L422 which puts it on the data parallel src rank only. But now we make it available everywhere.

I've fixed the comment here:  f50f5421ed8d291f9cdac98bfa3d61824b965b6f